csrc/multidevice/executor.cpp

```diff
@@ -155,12 +155,13 @@ void MultiDeviceExecutor::postKernel(SegmentedGroup* group) {
 
   // Compile the group and execute it with FusionExecutor
   // Check if the executor has been cached. If not, create and cache it
-  if (fe_.find(group) == fe_.end()) {
-    fe_.emplace(group, std::make_unique<FusionExecutor>());
-    fusions_.emplace(group, staged_fusion_->makeFusion(group));
-    fe_[group]->compileFusion(fusions_.at(group).get(), group_input_IValues);
+  if (fec_.find(group) == fec_.end()) {
+    fec_.emplace(
+        group,
+        std::make_unique<FusionExecutorCache>(
+            staged_fusion_->makeFusion(group), 0, auto_schedule_));
```

Comment:
Is `auto_schedule_` here always false?

Reply:
not necessarily, but for now we only have multidevice tests that set it to false. This is because we first need a patch on the scheduler to support multi-device. We are gonna open PRs soon about that

Reply:
see https://github.com/samnordmann/Fuser/tree/staged_reduction_tests_upstream for an example where we turn auto-scheduling on combined with manual multi-device scheduling. I will open that pr once this one is merged